jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java

```diff
@@ -187,39 +184,37 @@ private void addParcelSupport(JDefinedClass jclass) {
      *            new class. This node may include a 'javaType' property which
      *            if present will override the fully qualified name of the newly
      *            generated class.
-     * @param _package
-     *            the package which may contain a new class after this method
-     *            call
+     * @param container
+     *            the class container (package or class) which may contain a new
+     *            class after this method call
      * @return a reference to a newly created class
      * @throws ClassAlreadyExistsException
      *             if the given arguments cause an attempt to create a class
      *             that already exists, either on the classpath or in the
      *             current map of classes to be generated.
      */
-    private JDefinedClass createClass(String nodeName, JsonNode node, JPackage _package) throws ClassAlreadyExistsException {
+    private JDefinedClass createClass(String nodeName, JsonNode node, JClassContainer container) throws ClassAlreadyExistsException {
 
         JDefinedClass newType;
 
         Annotator annotator = ruleFactory.getAnnotator();
 
-        try {
-            if (node.has("existingJavaType")) {
-                String fqn = substringBefore(node.get("existingJavaType").asText(), "<");
-
-                if (isPrimitive(fqn, _package.owner())) {
-                    throw new ClassAlreadyExistsException(primitiveType(fqn, _package.owner()));
-                }
+        if (node.has("existingJavaType")) {
+            String fqn = substringBefore(node.get("existingJavaType").asText(), "<");
 
-                JClass existingClass = resolveType(_package, fqn + (node.get("existingJavaType").asText().contains("<") ? "<" + substringAfter(node.get("existingJavaType").asText(), "<") : ""));
-                throw new ClassAlreadyExistsException(existingClass);
+            if (isPrimitive(fqn, container.owner())) {
+                throw new ClassAlreadyExistsException(primitiveType(fqn, container.owner()));
             }
 
-            boolean usePolymorphicDeserialization = annotator.isPolymorphicDeserializationSupported(node);
```

Comment:
It seems as this branch had no difference in outcome from the other one, hence it was removed.